lib/wallet-e2e/src/Cardano/Wallet/Spec/Interpreters/Effectfully.hs

```diff
@@ -0,0 +1,62 @@
+module Cardano.Wallet.Spec.Interpreters.Effectfully
+    ( Story
+    , story
+    ) where
+
+import qualified Effectful.Error.Static as E
+
+import Cardano.Wallet.Spec.Effect.Assert
+    ( FxAssert, runAssertError )
+import Cardano.Wallet.Spec.Effect.Http
+    ( FxHttp, runHttpClient )
+import Cardano.Wallet.Spec.Effect.Query
+    ( FxQuery, runQuery )
+import Cardano.Wallet.Spec.Effect.Random
+    ( FxRandom, runRandom )
+import Cardano.Wallet.Spec.Effect.Trace
+    ( FxTrace, recordTraceLog, runTracePure )
+import Effectful
+    ( Eff, IOE, runEff )
+import Effectful.Fail
+    ( Fail, runFailIO )
+import Prelude hiding
+    ( Show, State, evalState, show )
+import System.Random
+    ( newStdGen )
+import Test.Syd
+    ( TestDefM, expectationFailure, it )
+import Text.Show
+    ( show )
+
+type Story a =
+    Eff
+        [ FxQuery
+        , FxHttp
+        , FxRandom
+        , Fail
+        , FxAssert
+        , FxTrace
+        , E.Error SomeException
```

Comment:
Can we get away without this E.Error effect if we reinterpret FxAssert instead of interpreting ?

Reply:
It would require handling errors locally in the Assert handler, which is not what it should be concerned with since there is already a dedicated Error handler anyway.